soc/nordic/nrf54h/gpd/gpd.c

```diff
@@ -181,11 +208,25 @@ int nrf_gpd_request(uint8_t id)
 		return -EIO;
 	}
 
-	sys_notify_init_spinwait(&client.notify);
+	if (k_is_pre_kernel()) {
+		sys_notify_init_spinwait(&client.notify);
+	} else {
+		sys_notify_init_callback(&client.notify, request_cb);
+		k_mutex_lock(&gpd_mgr->lock, K_FOREVER);
```

Comment:
The order should be reversed here, so that the mutex also protects the `client.notify` structure from modification while the request is in progress.

Reply:
client is local to the call, no?

Reply:
ah, right, missed that